src/compiler/checker.ts

```diff
@@ -11546,7 +11546,7 @@ namespace ts {
                 if (isBindingPattern(declaration.parent)) {
                     const parentDeclaration = declaration.parent.parent;
                     const name = declaration.propertyName || declaration.name;
-                    if (isVariableLike(parentDeclaration) &&
+                    if (parentDeclaration.kind !== SyntaxKind.BindingElement &&

```

Comment:
`BindingElement` has no property `type`. Previously it was implicitly excluded with the condition on the next line. Now, with more specific type information, that line would not type check any more without excluding `BindingElement` first.